include/xgboost/unixdefs.h

```diff
@@ -0,0 +1,14 @@
+#if defined(UNIXDEFS_H) && !defined(_MSC_VER) && \
+    !defined(__MINGW32__) && !defined(__CYGWIN__) && \
+    !defined(_WIN32) && !defined(__WINDOWS__) && \
+    !defined(__MUSL__)
+#define UNIXDEFS_H
+
+#define _GNU_SOURCE
+#include <features.h>
```

Comment:
Could you please share why we need this header?

Reply:
Feature flags. Can put it in another header if you think it works better semantically there.